src/main/java/uk/gov/nationalarchives/pdi/step/jena/shacl/JenaShaclDialog.java

```diff
@@ -0,0 +1,346 @@
+package uk.gov.nationalarchives.pdi.step.jena.shacl;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.ScrolledComposite;
+import org.eclipse.swt.events.*;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.layout.FormAttachment;
+import org.eclipse.swt.layout.FormData;
+import org.eclipse.swt.layout.FormLayout;
+import org.eclipse.swt.widgets.*;
+import org.pentaho.di.core.plugins.PluginInterface;
+import org.pentaho.di.core.plugins.PluginRegistry;
+import org.pentaho.di.core.plugins.StepPluginType;
+import org.pentaho.di.i18n.BaseMessages;
+import org.pentaho.di.trans.TransMeta;
+import org.pentaho.di.trans.step.BaseStepMeta;
+import org.pentaho.di.trans.step.StepDialogInterface;
+import org.pentaho.di.ui.core.ConstUI;
+import org.pentaho.di.ui.core.FormDataBuilder;
+import org.pentaho.di.ui.core.gui.GUIResource;
+import org.pentaho.di.ui.core.widget.ComboVar;
+import org.pentaho.di.ui.core.widget.TextVar;
+import org.pentaho.di.ui.trans.step.BaseStepDialog;
+
+import static uk.gov.nationalarchives.pdi.step.jena.Util.isNotEmpty;
+import static uk.gov.nationalarchives.pdi.step.jena.serializer.JenaSerializerStepMeta.DEFAULT_FILENAME;
+
+public class JenaShaclDialog extends BaseStepDialog implements StepDialogInterface {
+
+    private static final Class<?> PKG = JenaShaclStepMeta.class; // for i18n purposes
+
+    private static final int MARGIN_SIZE = 15;
+    private static final int LABEL_SPACING = 5;
+    private static final int ELEMENT_SPACING = 10;
+
+    private static final int LARGE_FIELD = 350;
+    private static final int MEDIUM_FIELD = 250;
+
+    private final JenaShaclStepMeta meta;
+
+    private Text wStepNameField;
+    private ComboVar wModelFieldCombo;
+    private TextVar wShapeFileTextField;
+    private FileDialog wBrowseFileDialog;
+    private boolean changed;
+
+    public JenaShaclDialog(final Shell parent, final Object in, final TransMeta transMeta, final String sname) {
+        super(parent, (BaseStepMeta) in, transMeta, sname);
+        meta = (JenaShaclStepMeta) in;
+    }
+
+    @Override
+    public String open() {
+        //Set up window
+        Shell parent = getParent();
```

Comment:
As a general code style, I prefer that any variable and/or parameter that can be `final` should be marked as final. This is definitely a style choice though - let's discuss

Reply:
I agree with you. Updated.